mempool/v1/reactor.go

```diff
@@ -269,6 +269,10 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 			if !success {
 				time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
 				continue
+			} else {
+				// record that we have sent the peer the transaction
+				// to avoid doing it a second time
+				memTx.SetPeer(peerID)
```

Comment:
lol what a simple change! makes sense to me.

the old mechanism seems _so weird_! Like, why even both setting the peer if they send you a transaction that you already have if they are setting the peer for original sender? Is the sole purpose to _attempt_ to keep txs in the mempool for longer? are there other tradeoffs?

Reply:
I think the original idea was to stop the transaction from being sent back to the peer who originally sent it